tests/test_encoding.py

```diff
@@ -220,6 +220,18 @@ def test_utf16_32(self):
         self._assert_encoding('utf-16', u"hi".encode('utf-16-be'), 'utf-16-be', u"hi")
         self._assert_encoding('utf-32', u"hi".encode('utf-32-be'), 'utf-32-be', u"hi")
 
+    def test_python_crash(self):
+        import random
+        from io import BytesIO
+        random.seed(42)
+        buf = BytesIO()
+        for i in range(150000):
+            buf.write(bytes([random.randint(0, 255)]))
```

Comment:
is it only me who worries about the randomness of this test case?
it looks like an use case for well known broken byte sequences or at the other end for [hypothesis](http://hypothesis.works/)

Reply:
It crashes when there are errors during decoding and sequence is above certain length (around 150kb). So random byte sequence is guaranteed to contain such errors. At the same time the random generator is initialised to a specific value, so it will generate the same sequence on every run.